### PR TITLE
docs(auroc): clarify per-batch logit/probability check in BinaryAUROC

### LINKED ISSUE #2195: Conversion from logits to probabilities happens on a batch by batch basis
## 🐛 Bug

I'm using `BinaryAUROC` and calling `.update(logits, y)` after each batch (`validation_step`), expecting the following behavior:

"If preds has values outside [0,1] range we consider the input to be logits and will auto apply sigmoid per element."

I noticed some discrepancies in the results, and after investigating realized that the rule above happens on a batch by batch basis rather than across the entire dataset (it happens in `_binary_precision_recall_curve_format`, which is called by `update`). 

Therefore, if some batch has all logits in [0, 1], then the conversion doesn't happen for that batch.

My use case has batch size = 1, so this happens often.

### Expected behavior

I'd expect that the conversion would happen consistently across the entire dataset (i.e., when calling `.compute()`). If not that, this behavior should be documented more prominently.

### Environment

```
torchmetrics==1.0.3
```

### PR DESCRIPTION
**Pull request by mohansree14**
## What does this PR do?

Fixes #2195

`BinaryAUROC.update()` (inherited from `BinaryPrecisionRecallCurve`) checks per batch whether `preds` looks like logits (values outside `[0, 1]`) and applies `sigmoid` if so. This check runs independently on every `update()` call rather than once across the full accumulated dataset, so with small batch sizes it's possible for some batches to be treated as probabilities and others as logits within the same run, producing inconsistent results.

Per the reporter's suggested resolution, this PR documents that behavior more prominently in `BinaryAUROC`'s class docstring rather than changing the underlying per-batch logic (which other metrics also rely on), to keep this a minimal, low-risk documentation fix. Happy to extend this to the base `BinaryPrecisionRecallCurve` docstring too, or take a different approach, if maintainers prefer.

## Before submitting
- [x] Was this discussed/agreed via a GitHub issue? (#2195)
- [x] Did you read the [contributor guideline](https://github.com/Lightning-AI/torchmetrics/blob/master/.github/CONTRIBUTING.md)?
- [x] Did you make sure your PR does only one thing, instead of bundling different changes together?
- [ ] Did you make sure to update the documentation with your changes? (n/a — this PR is the doc update)
- [ ] Did you write any new necessary tests? (n/a — documentation-only change, no behavior change)
- [ ] Did you verify new and existing tests pass locally?
- [ ] Did you list all the breaking changes introduced by this pull request?

This is my first contribution to this repo — feedback welcome.

### CODE REVIEW

**Comment by mohansree14** [comment]
Gentle ping on this one. CI is green and it's a docs-only change: a note in the `BinaryAUROC` docstring that the logit/probability check runs per `update()` call, as suggested in #2195. If you'd rather it go on `BinaryPrecisionRecallCurve` too, or you prefer a different fix, I'm happy to rework it.
